core/src/main/java/org/ironjacamar/core/connectionmanager/pool/dflt/DefaultManagedConnectionPool.java

```diff
@@ -100,11 +110,25 @@ public DefaultManagedConnectionPool(DefaultPool pool, Credential credential)
     */
    public ConnectionListener getConnectionListener() throws ResourceException
    {
+
       long timestamp = System.currentTimeMillis();
       while (System.currentTimeMillis() - timestamp <= pool.getConfiguration().getBlockingTimeout())
       {
-         for (ConnectionListener cl : listeners)
+         final Iterator<ConnectionListener> listenersIterator;
+         if (pool.isFIFO() && credential.equals(pool.getPrefillCredential()))
```

Comment:
Take this check to the constructor (AbstractMCP), as it is expensive, and set a protected boolean